recipe/meta.yaml

```
@@ -1,19 +1,19 @@
 {% set name = "pennylane-lightning-gpu" %}
-{% set version = "0.32.0" %}
-
+{% set version = "0.33.1" %}
+{% set build_num = 0 %}
 
 package:
   name: {{ name }}
   version: {{ version }}
 
 source:
-  url: https://github.com/PennyLaneAI/{{ name }}/archive/refs/tags/v{{ version }}.tar.gz
-  sha256: 1ef601a05fd231863dbf2e0f0f77fa8f7ffc4f3994426ae3fd8ebec542cdb287
+  url: https://github.com/PennyLaneAI/pennylane-lightning/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: 878f63cd1afadd52386b1aca9c0e3fb0a097b64ce8e347b325ebc7cac722e5e0
 
 build:
-  number: 0
-  skip: true  # [(not linux) or (cuda_compiler_version not in ("11.2", "12.0"))]
-  script: {{ PYTHON }} -m pip install . --no-deps -vv
+  number: {{ build_num }}
+  skip: true  # [(not linux) or (cuda_compiler_version not in ("11.8", "12.0"))]
+  script: PL_BACKEND=lightning_gpu {{ PYTHON }} -m pip install . --no-deps -vv
```

AFAIK (@vincentmr should correct me if I am wrong) pny doesn't care which CUDA minor version is in use, and to allow that at run time we need to relax the constraint imposed by the CUDA compiler:
```suggestion
  script: PL_BACKEND=lightning_gpu {{ PYTHON }} -m pip install . --no-deps -vv
  ignore_run_exports_from:
    - {{ compiler('cuda') }}
```
Note that the constraint on the CUDA version at run time was already added via pinning on `cuda-version`.

---

Should we add `cudatoolkit` on CUDA 11 then?

---

Those come with [`custatevec`](https://github.com/conda-forge/cuquantum-feedstock/blob/199b833ad118a7a7f3dfd925cc1d6df4c76f6ada/recipe/meta.yaml#L94-L95) but it wouldn't hurt to double check, especially with CUDA 12 new layout.

---

@leofang You're right. There isn't much CUDA code in there. As long as we can execute cuQuantum (cuStateVec in fact), cuBLAS and cuSPARSE I think we're good. We currently have `cuda_compiler_version not in ("11.2", "12.0")`, but I don't remember why those two versions. So we can probably relax that to `startswith` either `11` or `12`. I think cuStateVec depends on cuBLAS, so it should pull it, but we may need cudatoolkit for cuSPARSE-CU11, no?

---

Yeah, so we had two `cuda_compiler_version` because we built pny with two different CUDA major versions (11 & 12), and this was the way in conda-forge to specify the CI infrastructure. I assume pny should work with both versions without any patch?

For cuSPARSE, 
- on CUDA 11, `custatevec` depends on `cudatoolkit` which would include cuSPARSE too
- on CUDA 12, we have cuBLAS/cuSPARSE listed in the host section, which would run-export to `run`.

So in both cases we should be good, but we can improve the CUDA 12 handling a bit. I'll take care of it.